core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -731,6 +731,10 @@ public boolean isCompatibleWithInstalledVersion() {
                 else if (current.isOlderThan(requiredVersion)) {
                     deps.add(depPlugin);
                 }
+                // JENKINS-34494 - or if the plugin is disabled, this will allow us to enable it
+                else if (!current.isEnabled()) {
```

Comment:
A plugin can be both outdated and disabled. It seems that in this case, it wouldn't be enabled?